nomad/server.go

```diff
@@ -356,6 +376,106 @@ func (s *Server) Leave() error {
 	return nil
 }
 
+// setupConsulSyncer creates Server-mode consul.Syncer which periodically
+// executes callbacks on a fixed interval.
+func (s *Server) setupConsulSyncer() error {
+	// The bootstrapFn callback handler is used to periodically poll
+	// Consul to look up the Nomad Servers in Consul.  In the event the
+	// server has been brought up without a `retry-join` configuration
+	// and this Server is partitioned from the rest of the cluster,
+	// periodically poll Consul to reattach this Server to other servers
+	// in the same region and automatically reform a quorum (assuming the
+	// correct number of servers required for quorum are present).
+	bootstrapFn := func() error {
+		// If there is a raft leader, do nothing
+		if s.raft.Leader() != "" {
+			return nil
+		}
+
+		// If the the number of Raft peers or Members in Serf is more
+		// than the min quorum, do nothing.
+		raftPeers, err := s.raftPeers.Peers()
+		minQuorum := (s.config.BootstrapExpect / 2) + 1
+		if err == nil && (len(raftPeers) >= minQuorum || len(s.Members()) >= minQuorum) {
```

Comment:
Remove: `|| len(s.Members()) >= minQuorum)` because that is the serf Members. That will make it so that if you have federated with other servers you could skip even if you do not have local raft peers